### PR TITLE
aiogoogle module used for async changes in gcp_chronicle

### PR DESCRIPTION
**Pull request by thangaraj-ramesh**
aiogoogle module used for async changes in gcp_chronicle

### CODE REVIEW

**Comment by baulus** [comment]
Awesome job, thank you!
I tested with couple of scenarios, mostly it runs well, with handling wrong credentials, wrong host.
However, server timeout is not handled. You can simulate by assigning connector options timeout value 1. The details are in the other comment. 
Also, delete_connector does not handle well a delete for a non existing search id. 
```
failed to map: {'message': "\n\n\n\nNot Found\n\nContent:\n{'code': 404,\n 'message': 'generic::not_found: rule with ID '\n            'ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691 could not be found',\n 'status': 'NOT_FOUND'}\n\nRequest URL:\nhttps://backstory.googleapis.com/v2/detect/rules/ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691"}

{'success': False, 'connector': 'gcp_chronicle', 'error': "gcp_chronicle connector error => \n\n\n\nNot Found\n\nContent:\n{'code': 404,\n 'message': 'generic::not_found: rule with ID '\n            'ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691 could not be found',\n 'status': 'NOT_FOUND'}\n\nRequest URL:\nhttps://backstory.googleapis.com/v2/detect/rules/ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691", 'code': 'unknown'}
```

It is happening because your result_connector deletes the search_id after getting the results. This may be good, however, a client (UDI workers for example) will also call [delete_connector](https://github.com/thangaraj-ramesh/stix-shifter/blob/90c34ad9eb605997ea4a721d73b28bb6b67c9cc9/stix_shifter_modules/gcp_chronicle/stix_transmission/delete_connector.py#L1) after getting the results, and that will crowd the logs with a bunch of 'code': 'unknown' logs.
Please, also handle this.

File: stix_shifter_modules/gcp_chronicle/stix_transmission/api_client.py [hunk 1]
```diff
@@ -161,12 +164,25 @@ async def delete_search(self, search_id):
...
+        make an api call using aiogoogle request module
+        :params
+        url(str) -> host url
+        method(str) -> GET, POST
+        data(json) -> query input
+        :return: response
+        """
+        async with self.http_client as client:
+            req = Request(url=url, method=method, data=data)
+            return await client.as_service_account(req, full_res=True)
```
Comment (baulus) [inline stix_shifter_modules/gcp_chronicle/stix_transmission/api_client.py:188]: we need to use timeout argument as well, when calling `as_service_account`
```
return await client.as_service_account(req, full_res=True, timeout=self.timeout)
``` 

The self.timeout you can get in the class init:
`self.timeout = connection['options'].get('timeout')`

Also, please test with `self.timeout = 1` and handle the timeout error.

Comment (thangaraj-ramesh) [reply stix_shifter_modules/gcp_chronicle/stix_transmission/api_client.py]: Thanks for your comments.
As per your suggestion timeout and search id exceptions are handled.

**Comment by thangaraj-ramesh** [comment]
> Awesome job, thank you! I tested with couple of scenarios, mostly it runs well, with handling wrong credentials, wrong host. However, server timeout is not handled. You can simulate by assigning connector options timeout value 1. The details are in the other comment. Also, delete_connector does not handle well a delete for a non existing search id.
> 
> ```
> failed to map: {'message': "\n\n\n\nNot Found\n\nContent:\n{'code': 404,\n 'message': 'generic::not_found: rule with ID '\n            'ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691 could not be found',\n 'status': 'NOT_FOUND'}\n\nRequest URL:\nhttps://backstory.googleapis.com/v2/detect/rules/ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691"}
> 
> {'success': False, 'connector': 'gcp_chronicle', 'error': "gcp_chronicle connector error => \n\n\n\nNot Found\n\nContent:\n{'code': 404,\n 'message': 'generic::not_found: rule with ID '\n            'ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691 could not be found',\n 'status': 'NOT_FOUND'}\n\nRequest URL:\nhttps://backstory.googleapis.com/v2/detect/rules/ru_7ce598ed-7c38-4c9a-bdca-e24a5024e691", 'code': 'unknown'}
> ```
> 
> It is happening because your result_connector deletes the search_id after getting the results. This may be good, however, a client will also call [delete_connector](https://github.com/thangaraj-ramesh/stix-shifter/blob/90c34ad9eb605997ea4a721d73b28bb6b67c9cc9/stix_shifter_modules/gcp_chronicle/stix_transmission/delete_connector.py#L1) after getting the results, and that will crowd the logs with a bunch of 'code': 'unknown' logs. Please, also handle this.



delete of invalid search id exception is handled.

**Review by baulus — APPROVED** [review]

**Pull request merged**